### DISCUSSION TITLE
Can we link from an exported pdf file to a local file in exported pdf file?

### DISCUSSION THREAD

**Discussion by 136s** [slidevjs/slidev#1104] [Q&A]
When I set up a hyperlink to a local file and output a pdf, the link starts on `http://localhost:12445`.
How do I embed a relative link to a local file in a pdf?

# Example (part of `slides.md`):

```md
---

# Test

This is a [local link](./README.md)

---
```

and export a pdf file by `slidev export`.

# Expected behavior

I am expecting a relative path from the pdf file.

![a screenshot of the pdf file where ./README.md is the link](https://github.com/slidevjs/slidev/assets/47920723/c5a8d942-765a-4318-937c-a11a93e2e484)

# Actual behavior (part of `slides-export.pdf`)

However, the link embedded in the pdf starts on localhost.

![a screenshot of the pdf file where http://localhost:12445/README.md is the link](https://github.com/slidevjs/slidev/assets/47920723/3a4cbeac-9706-4b01-b23b-1608fd058913)

**Comment by JPustkuchen** [comment]
Don't think that's a question for slidev, just take a look if PDF's allow that. Perhaps Google knows?

**Comment by 136s** [comment]
Thanks for your reply!
Now that I understand that it is not a slidev question, I decided to fix it using [PyMuPDF](https://github.com/pymupdf/PyMuPDF) after generating the pdf.

Here is the [code](https://gist.github.com/136s/4a8baef9457fac067b9151b0b3a1d820) to fix it for anyone who comes to this page encountering the same problem.
